### PR TITLE
Fix HM1 code review points

### PR DESCRIPTION
**Pull request by laroself1**

### CODE REVIEW

File: src/storage/ArrayStorage.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+public class ArrayStorage {
+    private static final Logger log = Logger.getLogger("ArrayStorage Logger");
+
+    private static final int MAX_STORAGE_SIZE = 100000;
+    private final Resume[] storage = new Resume[MAX_STORAGE_SIZE];
+    private int size;
+
+    public void clear() {
+        Arrays.fill(storage, null);
```
Comment (den-ko) [inline src/storage/ArrayStorage.java]: You don't need to fill all the array with null values

File: src/storage/ArrayStorage.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            log.warning(String.format("Delete impossible. Resume(uuid=%s) is not present in storage%n", uuid));
+            return;
+        }
+        int indexToDelete = getResumeIndexById(uuid);
+        storage[indexToDelete] = storage[size - 1];
+        storage[size - 1] = null;
+        size--;
+    }
+
+    public Resume get(String uuid) {
```
Comment (den-ko) [inline src/storage/ArrayStorage.java:62]: Same, you iterate twice to find a resume, it's too expensive
It looks like you have StackOverflowError here. Have you run the application?

File: src/main/MainArray.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    public static void main(String[] args) {
+        Resume r1 = new Resume("uuid1");
+        Resume r2 = new Resume("uuid2");
+        Resume r3 = new Resume("uuid3");
+
+        arrayStorage.save(r1);
+        arrayStorage.save(r2);
+        arrayStorage.save(r3);
+
+        arrayStorage.update(r1);
```
Comment (den-ko) [inline src/main/MainArray.java]: How can you see that the update method actually works? Consider adding something to Resume, i.e. title, and update it

File: src/storage/ArrayStorage.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        }
+        if (isPresent(r.getUuid())) {
+            log.warning(String.format("Resume(uuid=%s) is already present in storage. It may be updated only.%n", r.getUuid()));
+        } else {
+            storage[size] = r;
+            size++;
+        }
+    }
+
+    public void update(Resume r) {
```
Comment (den-ko) [inline src/storage/ArrayStorage.java:36]: You iterate twice to find a resume, it's too expensive

File: src/storage/ArrayStorage.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        Objects.requireNonNull(r);
+        if (isPresent(r.getUuid())) {
+            storage[getResumeIndexById(r.getUuid())] = r;
+        } else {
+            log.warning(String.format("Resume(uuid=%s) is not present in storage. Please, add it.%n", r.getUuid()));
+        }
+    }
+
+    public void delete(String uuid) {
```
Comment (den-ko) [inline src/storage/ArrayStorage.java:46]: Same, you iterate twice to find a resume, it's too expensive

File: src/storage/ArrayStorage.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    public int size() {
+        return size;
+    }
+
+    private boolean isNotEmpty(String uuid) {
+        return uuid != null && !uuid.isEmpty();
+    }
+
+    private boolean isPresent(String uuid) {
+        return get(uuid) != null;
```
Comment (den-ko) [inline src/storage/ArrayStorage.java]: It's a bad idea to create cyclic dependencies

File: src/storage/ArrayStorage.java [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        return storage[getResumeIndexById(uuid)];
+    }
+
+    private int getResumeIndexById(String uuid) {
+        for (int i = 0; i < size; i++) {
+            if (uuid.equals(storage[i].getUuid())) {
+                return i;
+            }
+        }
+        return -1;
```
Comment (den-ko) [inline src/storage/ArrayStorage.java]: Magic numbers should be explanatory

**Review by den-ko — COMMENTED** [review]
Done

File: src/storage/ArrayStorage.java (hunk 1 above)
Comment (laroself1) [reply src/storage/ArrayStorage.java]: Left as is don't know how to resolve. I can recreate the array only

File: src/storage/ArrayStorage.java (hunk 2 above)
Comment (laroself1) [reply src/storage/ArrayStorage.java:62]: It was an error, have not checked after another refactoring, Fixed

File: src/main/MainArray.java (hunk 3 above)
Comment (laroself1) [reply src/main/MainArray.java]: Done

File: src/storage/ArrayStorage.java (hunk 4 above)
Comment (laroself1) [reply src/storage/ArrayStorage.java:36]: Fixed

File: src/storage/ArrayStorage.java (hunk 6 above)
Comment (laroself1) [reply src/storage/ArrayStorage.java]: Fixed

File: src/storage/ArrayStorage.java (hunk 5 above)
Comment (laroself1) [reply src/storage/ArrayStorage.java:46]: Fixed

File: src/storage/ArrayStorage.java (hunk 7 above)
Comment (laroself1) [reply src/storage/ArrayStorage.java]: Done

**Pull request merged**